fpmsyncd/routesync.cpp

```diff
@@ -886,6 +967,62 @@ bool RouteSync::getIfName(int if_index, char *if_name, size_t name_len)
     return true;
 }
 
+/*
+ * Get next hop List
+ * @arg route_obj     route object
+ *
+ * Return concatenation of NHs: nh0 + "," + nh1 + .... + "," + nhN
+ */
+string RouteSync::getNextHopList(struct rtnl_route *route_obj)
+{
+    string result = "";
+
+    for (int i = 0; i < rtnl_route_get_nnexthops(route_obj); i++)
+    {
+        struct rtnl_nexthop *nexthop = rtnl_route_nexthop_n(route_obj, i);
+        struct nl_addr *addr = NULL;
+
+        if ((addr = rtnl_route_nh_get_newdst(nexthop)))
```

Comment:
what is the API returning? It's not clear whether it returns a label stack or just a single label. The code seems to assume a single label.

Reply:
rtnl_route_nh_get_newdst() returns full label stack in nl_addr struct.  The code does not assume a single label.  nl_addr2str formats the full label stack into a single string with '/' between each label.  '+' is used to separate label stack from IP gateway (ie, via) information in NH.

Reply:
MAX_ADDR_SIZE  is defined as 64, which is for one IPv6 address. Using MAX_ADDR_SIZE, there can be up to 16 labels. Is there a limit in label stack depth. Most ASICs support up to 4. What's the supported stack depth. 
 
 There should be an error thrown if label stack depth received is more than supported.

Reply:
"nl_addr2str formats the full label stack into a single string with '/' between each label"

This doesn't seem to be true, at least from the code I saw.

can you paste can example of the output run from your API.

Reply:
nl_addr2str() in libnl3/libnl-3.5.0/lib/addr.c calls mpls_ntop() in libnl3/libnl-3.5.0/lib/mpls.c.  It is definitely iterating through the full array of labels and inserting '/' between for format.